src/Commands/BuildToolsCommand.php

```diff
@@ -39,6 +43,413 @@ public function __construct()
+        // to Pantheon. We cannot recover the token for the currently-authenticated
+        // user, because security. Unfortunately, there is no API to generate
+        // a new machine token. Otherwise, we'd just do that, which would have
+        // the side benefit of providing a dedicated token for each site.
+        $terminus_token = getenv('TERMINUS_TOKEN');
+        $github_token = getenv('GITHUB_TOKEN');
+        $circle_token = getenv('CIRCLE_TOKEN');
+
+        if (empty($terminus_token)) {
+            throw new TerminusException("Please generate a Pantheon machine token, as described in https://pantheon.io/docs/machine-tokens/. Then run: \n\nexport TERMINUS_TOKEN=my_machine_token_value");
+        }
+
+        if (empty($github_token)) {
+            throw new TerminusException("Please generate a GitHub personal access token token, as described in https://help.github.com/articles/creating-an-access-token-for-command-line-use/. Then run: \n\nexport GITHUB_TOKEN=my_personal_access_token_value");
+        }
+
+        if (empty($circle_token)) {
+            throw new TerminusException("Please generate a Circle CI personal API token token, as described in https://circleci.com/docs/api/#authentication. Then run: \n\nexport CIRCLE_TOKEN=my_personal_api_token_value");
+        }
+
+        $this->log()->notice('Create GitHub project {org}/{target} from {src}', ['src' => $source, 'org' => $github_org, 'target' => $target]);
+        // Create the GitHub repository
+        list($target_project, $siteDir) = $this->createGitHub($source, $target, $github_org, $github_token);
+
+        $this->log()->notice('Create ssh key pair');
+        // Create an ssh key pair dedicated to use in these tests.
+        list($publicKey, $privateKey) = $this->createSshKeyPair($git_email, $site_name . '-key');
+        $this->addPublicKeyToPantheonUser($publicKey);
+
+        // Look up our upstream.
+        $upstream = $this->autodetectUpstream($siteDir);
+
+        // Push our site to Pantheon.
+        $this->createPantheonSite($site_name, $siteDir, $label, $team, $upstream);
+
+        // Look up the site UUID for the Pantheon dashboard link
+        $site = $this->getSite($site_name);
+        $siteInfo = $site->serialize();
+        $site_uuid = $siteInfo['id'];
+
+        $this->log()->notice('Created a new Pantheon site with UUID {uuid}', ['uuid' => $site_uuid]);
+
+        // Create a new README file to point to this project's Circle tests and the dev site on Pantheon
+        $circleBadge = "[![CircleCI](https://circleci.com/gh/$source.svg?style=svg)](https://circleci.com/gh/{$target_project})";
+        $pantheonBadge = "[![Pantheon {$target}](https://img.shields.io/badge/pantheon-{$target}-yellow.svg)](https://admin.dashboard.pantheon.io/sites/{$site_uuid}#dev/code)";
+        $siteBadge = "[![Dev Site {$target}](https://img.shields.io/badge/site-{$target}-blue.svg)](http://dev-{$target}.pantheonsite.io/)";
```

Comment:
Two of the generated shields failed here: https://github.com/stevectorbot/stevectorbot-40